### PR TITLE
#5 user foto management

### PR DESCRIPTION
**Pull request by BjornVanBruchem**
# Description

Please include a summary of the change, which issue is fixed and the context if the new feature. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant and add new options if necessary. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: management/src/pages/RejectedphotoManagement.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        });
+        const body = JSON.stringify({ id: item.userImageId });
+
+        await fetch(process.env.REACT_APP_BACKEND_ROOT_URL + '/api/user-image/delete', {
+            method: 'POST',
+            body: body,
+            headers: headers
+        });
+
+        const index = images.findIndex(i => i.description === item.userImageId);
```
Comment (xandermonk) [inline management/src/pages/RejectedphotoManagement.js:89]: Waarom wordt hier de description vergeleken met de ID? Lijkt me niet de juiste aanpak.

File: management/src/pages/photoManagement.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    const refreshData = () => {
+        if (galleryImages) {
+            galleryImages.forEach(i => {
+                if (i.approvalStatus === 'pending') {
+                    const data = `data:${i.image.contentType};base64,${Buffer.from(i.image.data)}`;
+                    images.push({
+                        original: data,
+                        thumbnail: data,
+                        srcSet: data,
+                        originalAlt: i.image.name,
```
Comment (xandermonk) [inline management/src/pages/photoManagement.js:33]:
```suggestion
                        originalAlt: i.image?.name,
```

File: management/src/pages/RejectedphotoManagement.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                        thumbnail: data,
+                        srcSet: data,
+                        originalAlt: i.image.name,
+                        thumbnailAlt: i.image.name,
+                        userImageId: i._id
+                    });
+                }
+            });
+        }
+        console.log()
```
Comment (xandermonk) [inline management/src/pages/RejectedphotoManagement.js:41]:
```suggestion
```
dit kan weg

File: management/src/pages/RejectedphotoManagement.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        if (galleryImages) {
+            galleryImages.forEach(i => {
+                if (i.approvalStatus === 'declined') {
+                    const data = `data:${i.image.contentType};base64,${Buffer.from(i.image.data)}`;
+                    images.push({
+                        original: data,
+                        thumbnail: data,
+                        srcSet: data,
+                        originalAlt: i.image.name,
+                        thumbnailAlt: i.image.name,
```
Comment (xandermonk) [inline management/src/pages/RejectedphotoManagement.js:35]:
```suggestion
                        thumbnailAlt: i.image?.name,
```
voor het geval image niet bestaat, voorkomt een runtime error

File: management/src/pages/photoManagement.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                        thumbnailAlt: i.image.name,
+                        userImageId: i._id
+                    });
+                }
+            });
+        }
+        console.log()
```
Comment (xandermonk) [inline management/src/pages/photoManagement.js:40]:
```suggestion
```
dit kan weg

File: management/src/pages/photoManagement.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        });
+        const body = JSON.stringify({ id: item.userImageId });
+
+        await fetch(process.env.REACT_APP_BACKEND_ROOT_URL + '/api/user-image/decline', {
+            method: 'POST',
+            body: body,
+            headers: headers
+        });
+
+        const index = images.findIndex(i => i.description === item.userImageId);
```
Comment (xandermonk) [inline management/src/pages/photoManagement.js:98]: Waarom wordt hier de description vergeleken met de ID? Lijkt me niet de juiste aanpak.

**Review by xandermonk — CHANGES_REQUESTED** [review]

**Review by Veaxx — APPROVED** [review]

**Review by GXZheng312 — APPROVED** [review]

**Pull request merged**